cleverpush/src/main/java/com/cleverpush/manager/SubscriptionManagerBase.java

```diff
@@ -129,6 +129,12 @@ public void onSuccess(String response) {
           JSONObject responseJson = new JSONObject(response);
           if (responseJson.has("id")) {
             String newSubscriptionId = responseJson.getString("id");
+            String oldSubscriptionId = sharedPreferences.getString(CleverPushPreferences.SUBSCRIPTION_ID, null);
```

Comment:
(Same like in iOS PR)
Can we do the detection if subscription ID changed inside CPChatView, instead of in CleverPush.m subscribed handler?
It would be better because we only need this information in the CPChatView Class.

We already have the `subscriptionId` variable in CPChatView and can compare with `getSubscriptionId` inside loadChat

Reply:
https://github.com/cleverpush/cleverpush-android-sdk/blob/master/cleverpush/src/main/java/com/cleverpush/manager/SubscriptionManagerBase.java#L132

In ChatView -> loadChat we pass the updated `subscriptionId`. we can not compare the `subscriptionId` in the loadChat method. 

Whenever we unsubscribe we remove all the data
```
public void clearSubscriptionData() {
    subscriptionId = null;
    SharedPreferences sharedPreferences = getSharedPreferences(getContext());
    SharedPreferences.Editor editor = sharedPreferences.edit();
    editor.remove(CleverPushPreferences.SUBSCRIPTION_ID);
    editor.remove(CleverPushPreferences.SUBSCRIPTION_LAST_SYNC);
    ...
    editor.commit();
    ...
  }
```

In Subscribe we update the new id with the old id. So we don't have an old id for comparison. At the time of subscription, before updating the new subscriptionId, we get the old subscriptionId and check with the response id whether both are the same or not.

<img width="822" alt="Screenshot 2023-07-25 at 11 08 07 AM" src="https://github.com/cleverpush/cleverpush-android-sdk/assets/129381063/f33be761-be03-4184-a317-3e4c98ab39a5">